src/dft/descriptor.cxx

```diff
@@ -17,60 +17,163 @@
 * SPDX-License-Identifier: Apache-2.0
 *******************************************************************************/
 
+#include "oneapi/mkl/detail/exceptions.hpp"
 #include "oneapi/mkl/dft/descriptor.hpp"
 
 namespace oneapi {
 namespace mkl {
 namespace dft {
 namespace detail {
 
+// Compute the default strides for a domain. Modifies real_strides and complex_strides
+// argument.
+template <domain dom>
+static void compute_default_strides(const std::vector<std::int64_t>& dimensions, int rank,
+                                    std::vector<std::int64_t>& input_strides,
+                                    std::vector<std::int64_t>& output_strides);
+
+template <>
+void compute_default_strides<domain::REAL>(const std::vector<std::int64_t>& dimensions, int rank,
+                                           std::vector<std::int64_t>& input_strides,
+                                           std::vector<std::int64_t>& output_strides) {
+    // Compute strides for CCE format.
```

Comment:
I'm still a bit unsure and confused about this.

If the transform is real -> complex then I would expect input_strides to have normal strides (as defined in compute_default_strides<domain::COMPLEX>). While the output_strides would have the values from this function (because of the CCE Format). Did you test this scenario?

I'm also unsure how the backward transform (complex to real) would work  in this scenario. Because the input and output would presumably be reversed (and hence the strides would also be reversed)

I think it might be a good idea to check that this scenarios give the expected results.

Reply:
That's a tough one, I've been looking at https://spec.oneapi.io/versions/1.2-rev-1/elements/oneMKL/source/domains/dft/config_params/strides.html#onemkl-dft-config-strides to understand how this stride should work. My take on this is:

* The strides should be the same for forward and backward transform.
* They represent the offset in number of elements (complex or real) and not in number of floats or doubles. It's not clear to me whether this is also true for the `dimensions` but I assume it is.
  * That means that to me the specialization for `domain::COMPLEX` is correct and the `domain::REAL` one should work the way as the `COMPLEX` one.

I'm really not sure about any of this too. I suppose it will be easier to figure out once we have tests we can compile.

Reply:
I've compared the imlementation against values given in adapted examples from Intel's closed-source MKLGPU library:
* `sp_complex_3d`: A complex domain 3d transform. The default values are used, and the DFT's results are checked. I think the default strides computed for complex are correct.
* Real:
  * `sp_real_3d`: A real-domain 3d inplace transform. This uses the default DFT config. The default values computed here match those explicitly set in the example. 
  * `sp_real_3d_outofplace`: A real-domain 3d outofplace transform. The default values computed here do not match those explicilty set in the example. I think this is correct since out-of-place transforms are not the default behaviour.
  *  I think the default strides computed for real are correct.

So I think the current stride implementation is correct.

@Rbiessy, the [C MKL](https://www.intel.com/content/www/us/en/develop/documentation/onemkl-developer-reference-c/top/fourier-transform-functions/fft-functions/configuration-settings/dfti-input-strides-dfti-output-strides.html#dfti-input-strides-dfti-output-strides) reference gives the impression this is incorrect for the CCE format used here. I've implemented the strides here according the the guidance for the CCE format.

Reply:
This has been revised to match the closed-source Intel MKL in [f4ec8e5](https://github.com/hjabird/oneMKL/pull/3/commits/f4ec8e57ad5129244e2f6f84dcc19e9fde8230ad).